src/System.Management/src/System/Management/ManagementDateTime.cs

```diff
@@ -117,113 +117,94 @@ public static DateTime ToDateTime(string dmtfDate)
+                    second = int.Parse(tempString,frmInt32);
                 }
                 tempString = dmtf.Substring(15, 6);
                 if (("******" != tempString)) 
                 {
-                    ticks = (System.Int64.Parse(tempString,(IFormatProvider)CultureInfo.InvariantCulture.GetFormat(typeof(System.Int64)))) * (System.TimeSpan.TicksPerMillisecond/1000);
+                    ticks = (long.Parse(tempString,(IFormatProvider)CultureInfo.InvariantCulture.GetFormat(typeof(Int64)))) * (TimeSpan.TicksPerMillisecond/1000);
                 }
-                if( year < 0 || month < 0 || day < 0 || hour < 0 || minute < 0 || second < 0 || ticks < 0)
+                tempString = dmtf.Substring(22, 3);
+                if (("***" != tempString)) 
                 {
-                    throw new System.ArgumentOutOfRangeException("dmtfDate");
+                    tempString = dmtf.Substring(21, 4);
+                    utcOffset = int.Parse(tempString,frmInt32);
                 }
 
+                if( year < 0 || month < 0 || day < 0 || hour < 0 || minute < 0 || second < 0 || ticks < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(dmtfDate));
+                }
             }
             catch
             {
-                throw new System.ArgumentOutOfRangeException("dmtfDate");
+                throw new ArgumentOutOfRangeException(nameof(dmtfDate));
             }
 
 
             // Construct a new System.DateTime object
-            datetime = new System.DateTime(year, month, day, hour, minute, second, millisec);
+            var datetime = new DateTime(year, month, day, hour, minute, second, 0, DateTimeKind.Utc);
             // Then add the ticks calculated from the microseconds
             datetime = datetime.AddTicks(ticks);
-            
-            // Adjust the UTC time to reflect the current zone time
-            System.TimeZone curZone = System.TimeZone.CurrentTimeZone;
-            System.TimeSpan tickOffset = curZone.GetUtcOffset(datetime);
-            long OffsetMins = tickOffset.Ticks / System.TimeSpan.TicksPerMinute;
-            
-            // Adjusting the DateTime for the current UTC
-            int UTCOffset =  0;
-            string tempString1 = dmtf.Substring(22, 3);
-            long OffsetToBeAdjusted = 0;
-            if (("***" != tempString1)) 
-            {
-                tempString1 = dmtf.Substring(21, 4);
-                try
-                {
-                    UTCOffset = System.Int32.Parse(tempString1,frmInt32);
-                }
-                catch
-                {
-                    throw new System.ArgumentOutOfRangeException();
-                }
-
-                OffsetToBeAdjusted = UTCOffset-OffsetMins;
-                
-                // We have to substract the minutes from the time
-                datetime = datetime.AddMinutes(OffsetToBeAdjusted * -1);
+            // Then subtruct offset in minutes
```

Comment:
nit: there is a typo 'subtruct' but I rather prefer a more generic comment, e.g.: "Adjust minutes according to UTC offset", since we actually don't know the signal.